### ISSUE TITLE
[bug] Star Trek Legacy does not work on Vista or newer even with openspy-client.

### ISSUE THREAD

**Issue by BlackAnt1968601851** [anzz1/openspy-client#14]
### Game name

Star Trek Legacy

### 🔍

- [x] The game is in the [supported games list](https://github.com/anzz1/openspy-client#supported-games-list).

### What happened?

When I try to join someone, it says "Connection failed", and the host's game just simply freezes when the client tries to join them.
The connection failed screen is shown below.
![Connection Failed](https://github.com/user-attachments/assets/dc7a1e30-c34f-4f2c-b695-c708aeb0a239)

You can see I am running the latest patch below
![Game Version 1 030](https://github.com/user-attachments/assets/b13b2c25-abfa-4b7e-a8cd-82a8fedf173c)

The Lan function doesn't detect servers but the internet option does detect
![Multiplayer List](https://github.com/user-attachments/assets/3c43eab4-46cf-4301-b335-8522f630c0d2)

Below will be the list of lan sessions
![Multiplayer List](https://github.com/user-attachments/assets/fd7760bd-4449-46c1-bc2c-272994c205d7)

### Game version

1.2/1.030

### Game release

CD / DVD

### OpenSpy DLL Name

dinput8.dll

### OpenSpy DLL Version

v1.0.78

### Additional game information

Both the client and the host are both running the same version.
using the same dll and both are set as DMZ on network.

**Comment by Ligushka** [comment]
I'm looking a bit into this. I am able to replicate the issue by hosting a LAN game and trying to connect using another instance of the game - the host freezes when the client queries for LAN games. The host's game seems to deadlock on an `EnterCriticalSection` in `NetworkManager.dll`'s `Network::DirectPlay::DirectPlayMessageHandler`. I don't believe it's related to `openspy-client`, but something with the game itself. I will try to figure it out.

Are you running the game on Windows 11 24H2?

**Comment by BlackAnt1968601851** [comment]
yes I am running windows 11 24H2 but it also happens on windows 10 and windows 7 too.

**Comment by anzz1** [comment]
Have you installed DirectPlay from 'Turn Windows features on or off'?

**Comment by BlackAnt1968601851** [comment]
Yeah, I make sure I have all the requirements since I do IT, and I know each game requires.
It doesn't work on anything newer than windows xp

**Comment by BlackAnt1968601851** [comment]
I have directplay already installed since the day I installed windows 11 to my desktop.
I had two laptops I tried connecting to each other through the lan and public options. but it continues to crash or freeze

I make sure I have requirements for all my games by installing all visual c++ redists from 2005 - 2022
I also usually install the Direct X June 2010 Redist as well.
Im not a developer so I can't trace the issue but it looks like the guy named Ligushka has replicated the problem.

![Image](https://github.com/user-attachments/assets/4bcc2456-1a14-4e82-b43a-cd77a87f1a8e)

**Comment by BlackAnt1968601851** [comment]
I found some new information, it seems to be something in the newer copies of windows that broke directplay in a way that prevents the game from connecting
because when I ran vista in a VM it didn't freeze up when trying to join it, but my windows 11 machine still fails to connect to it.

**Comment by BlackAnt1968601851** [comment]
The game still freezes in Windows Vista, it seems to be a issue with the game's code or something because I ran it in vista and it did the same deadlock

**Comment by BlackAnt1968601851** [comment]
The game seems to only work on Windows XP, Vista or newer doesn't work.

**Comment by BlackAnt1968601851** [comment]
![Image](https://github.com/user-attachments/assets/dfe0a79c-28ad-4cd2-8e05-5fd224755c78)

when I use the openspy-client dll, the game crashes when trying to join.

**Comment by BlackAnt1968601851** [comment]
if you guys want to play multiplayer it is easier to set up a VM with Windows XP and remove the openspy-client dll so it doesn't crash when using the lan option to join.

**Comment by BlackAnt1968601851** [comment]
the deadlock seems to be caused by openspy-client somehow because removing it stopped the deadlock.

**Comment by BlackAnt1968601851** [comment]
I think the deadlock is caused by the patch to the stlegacy_disable_net_voice because when that is patched on xp it does the same freeze but without the openspy-client dll, it doesn't freeze.

**Comment by BlackAnt1968601851** [comment]
I manually patched the dll directly using hxd editor to disable net voice and the freeze happens.

**Comment by BlackAnt1968601851** [comment]
anzz1 is now aware of the issue, I got in contact with him on discord and he was able to replicate the issue.

**Comment by BlackAnt1968601851** [comment]
we did some testing and got it to work.

**Comment by BlackAnt1968601851** [comment]
I am closing this since we got it fixed now.

**Comment by anzz1** [comment]
@Ligushka it's fixed now in the latest release.
It was a stupid mistake on my part, I accidentally skipped the `LeaveCriticalSection` in `Network::DirectPlay::GameCreate` instead of just skipping the call to `Network::Lobby::GetVoiceSettings()->0x40()` what I meant to, leaving the critical section locked forever. Correcting this fixed the issue.
